src/d/actor/d_a_leaflift.cpp

```diff
@@ -4,98 +4,348 @@
+        if (strcmp("j_happa", jointName->getName(i)) == 0) {
+            mpModel->getModelData()->getJointNodePointer(i)->setCallBack(&nodeCallBack);
+            break;
+        }
+    }
+    mTargetRotation = ZeroQuat;
+    mCurrentRotation = mTargetRotation;
+    if ((s32)daLlift_prm::getPos(this) == 1) {
+        current.pos.y =  home.pos.y + m_height;
+    }
+    else {
+        if (mEmitter3) {
+            mEmitter3->playCreateParticle();
+        }
+    }
+    m43F = 0x1e;
+    mpModel->calc();
+    setMoveBGMtx();
+    dComIfG_Bgsp()->Regist(mpBgW, this);
+    set_mtx();
+    mpBgW->Move();
 }
 
 /* 00000634-00000760       .text _create__9daLlift_cFv */
 s32 daLlift_c::_create() {
-    /* Nonmatching */
+    int res;
+    fopAcM_SetupActor(this, daLlift_c);
+    res = dComIfG_resLoad(&mPhs, m_arcname);
+    if (res == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0xf40)) {
+            return cPhs_ERROR_e;
+        }
+        else {
+            CreateInit();
+        }
+    }
+    return res;
 }
 
 /* 00000918-000009C4       .text nodeCallBack__FP7J3DNodei */
-static BOOL nodeCallBack(J3DNode*, int) {
-    /* Nonmatching */
+static BOOL nodeCallBack(J3DNode* node, int idx) {
+    if (!idx) {
+        J3DJoint* joint = (J3DJoint*)node;
+        s32 jntNo = joint->getJntNo();
+        J3DModel* model = j3dSys.getModel();
+        daLlift_c* i_this = (daLlift_c*)model->getUserArea();
+        if (i_this) {
+            mDoMtx_stack_c::copy(model->getAnmMtx(jntNo));
+            mDoMtx_stack_c::quatM(&i_this->mCurrentRotation);
+            cMtx_copy(mDoMtx_stack_c::get(), J3DSys::mCurrentMtx);
+            model->setAnmMtx(jntNo, mDoMtx_stack_c::get());
+            cMtx_copy(mDoMtx_stack_c::get(), i_this->mBGMtx);
+        }
+    }
+    return TRUE;
 }
 
 /* 000009C4-00000CAC       .text rideCallBack__FP4dBgWP10fopAc_ac_cP10fopAc_ac_c */
-void rideCallBack(dBgW*, fopAc_ac_c*, fopAc_ac_c*) {
-    /* Nonmatching */
+static void rideCallBack(dBgW* param1, fopAc_ac_c* i_this, fopAc_ac_c* i_other) {
+    f32 tiltFactor;
+    f32 offsetMagnitude;
+    daLlift_c* param2 = (daLlift_c*)i_this;
+    daLlift_c* param3 = (daLlift_c*)i_other;
+
+    cXyz posOffset = param3->current.pos - param2->current.pos;
+    
+    tiltFactor = 2.0f;
+    if (fopAcM_GetName(param3) == PROC_PLAYER) {
+        param2->m43C = TRUE;
+        param2->m43E = TRUE;
+        posOffset = posOffset.outprod(up_vec);
+        mDoMtx_stack_c::YrotS(-param2->current.angle.y);
+        mDoMtx_stack_c::multVec(&posOffset, &posOffset);
+        if (param2->m43D) {
+            posOffset = param3->current.pos;
+            posOffset.x += 200.0f;
+            posOffset = posOffset.outprod(up_vec);
+            mDoMtx_stack_c::YrotS(-param2->current.angle.y);
+            mDoMtx_stack_c::multVec(&posOffset, &posOffset);
+            tiltFactor = cM_ssin((param2->mGlideFrameCounter & 0x1F) << 11) * 2.0f * 0.25f * (cM_rndFX(0.2f) + 1.0f);
```

Comment:
This looks like an optimization too, come to think of it. Try something like:
```suggestion
            s16 angle = (param2->mGlideFrameCounter % 32) * (0x10000 / 32);
            tiltFactor = cM_ssin(angle * 2.0f * 0.25f * (cM_rndFX(0.2f) + 1.0f);
```
If I'm understanding this math right, I think this angle is basically cycling over all 360 degrees (0x10000 cSAngles) over the course of every 32 frames?

Reply:
That is the correct equivalent function but the code doesn't match when doing that.

Reply:
Ok, for future reference it turned out the reason my suggestion didn't match was because the 32 literal was signed (not sure why that matters...). Using `32U` instead of `32` matches:
```cpp
s16 angle = (i_this->mActorLifetimeFrameCount % 32U) * (0x10000 / 32);
```